### PR TITLE
Follow or Unfollow an user

### LINKED ISSUE #35: Follow an user
- [x] The user can follow to other users.
- [x] When a user follows someone, their followers increase by 1 (or decrease by 1 in the opposite situation).

### PR DESCRIPTION
**Pull request by angel23v**
## Quick Info
Functionality to follow or unfollow an user. 
Now an user can have followers or followings users

## Screenshots
Screenshot from 2024-03-26 01-06-55
Screenshot from 2024-03-26 01-06-48

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -23,12 +23,14 @@ def create
...
   end
 
   def edit
-    unless current_user == @post.user
-      redirect_to root_path, alert: 'You are not authorized to edit this post.'
-    end
+    return if current_user == @post.user
+
+    redirect_to root_path, alert: 'You are not authorized to edit this post.'
   end
```
Comment (MauricioNC) [inline app/controllers/posts_controller.rb]: Why do you remove the past code? What is the purpose of making a return?

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -49,9 +51,7 @@ def update
...
 
   def destroy
-    unless current_user == @post.user
-      redirect_to root_path, alert: 'You are not authorized to delete this post.'
-    end
+    redirect_to root_path, alert: 'You are not authorized to delete this post.' unless current_user == @post.user
```
Comment (MauricioNC) [inline app/controllers/posts_controller.rb]: If you will restrict the access to some actions or views, you should send a status code like:

`redirect_to root_path, alert: 'You are not authorized to delete this post.', status: :unauthorized unless current_user == @post.user`

By this way, the server is telling the client that this action is unauthorized

File: app/javascript/controllers/hello_controller.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
-import { Controller } from "@hotwired/stimulus"
+import { Controller } from "@hotwired/stimulus";
 
 export default class extends Controller {
   connect() {
-    this.element.textContent = "Hello World!"
+    this.element.textContent = "Hello World!";
   }
 }
```
Comment (MauricioNC) [inline app/javascript/controllers/hello_controller.js]: Remove code and files that you don't need

File: app/views/devise/registrations/edit.html.erb [hunk 4]
```diff
@@ -3,7 +3,10 @@
 <%= form_for(@user, url: user_path(@user), method: :patch, html: { class: "my-form" }) do |form| %>
    
    <div class="field">
+<<<<<<< HEAD
+=======
     <%= image_tag image_profile?([10, 10]) %>
+>>>>>>> fc90c211047c627e9df511b389467131841725e2
     <%= form.label :name %>
```
Comment (MauricioNC) [inline app/views/devise/registrations/edit.html.erb]: What is this?

File: app/views/devise/sessions/new.html.erb [hunk 5]
```diff
@@ -28,8 +28,8 @@
   <div class="links">
     <%= f.submit "Log in" , class:"btn"%>
 
-    <%= render "devise/shared/links" %>
   </div>
 <% end %>
+    <%= render "devise/shared/links" %>
 
 </div>
```
Comment (MauricioNC) [inline app/views/devise/sessions/new.html.erb]: Remove empty lines

**Review by MauricioNC — CHANGES_REQUESTED** [review]

File: app/controllers/posts_controller.rb [hunk 6]
```diff
@@ -23,12 +23,14 @@ def create
...
+    @follow = User.where(user_id: current_user.id)
   end
 
   def edit
-    unless current_user == @post.user
-      redirect_to root_path, alert: 'You are not authorized to edit this post.'
-    end
+    return if current_user == @post.user
+
+    redirect_to root_path, alert: 'You are not authorized to edit this post.'
```
Comment (MauricioNC) [inline app/controllers/posts_controller.rb]: Same here, send a status code with the response

**Review by MauricioNC — CHANGES_REQUESTED** [review]

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (angel23v) [reply app/controllers/posts_controller.rb]: jajja ahhh Idk why that is again, maybe on a PR I guess let me solve it

File: app/controllers/posts_controller.rb [hunk 7]
```diff
@@ -23,20 +23,18 @@ def create
...
     @user_likes = current_user.likes.where(post_id: @posts.map(&:id)).index_by(&:post_id)
+    @users = User.all
+    @follow = User.where(user_id: current_user.id)
   end
 
   def edit
-    unless current_user == @post.user
-      redirect_to root_path, alert: 'You are not authorized to edit this post.'
-    end
+    redirect_to root_path, alert: 'You are not authorized to edit this post.' unless current_user == @post.user
```
Comment (MauricioNC) [inline app/controllers/posts_controller.rb]: Send a status code here too

File: app/controllers/posts_controller.rb [hunk 8]
```diff
@@ -23,20 +23,18 @@ def create
...
   end
 
   def show; end
 
   def update
-    unless current_user == @post.user
-      redirect_to root_path, alert: 'You are not authorized to update this post.'
-    end
+    redirect_to root_path, alert: 'You are not authorized to update this post.' unless current_user == @post.user
```
Comment (MauricioNC) [inline app/controllers/posts_controller.rb]: Send a status code here too

**Review by MauricioNC — CHANGES_REQUESTED** [review]

File: app/controllers/followers_controller.rb [hunk 9]
```diff
@@ -0,0 +1,30 @@
+class FollowersController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    @user = User.find(params[:user_id])
```
Comment (JuanequeX) [inline app/controllers/followers_controller.rb]: Add this line in a private method call it `set_user` and sent it to a before action only to `create` and `destroy` method

**Review by JuanequeX — CHANGES_REQUESTED** [review]

**Review by JuanequeX — APPROVED** [review]

**Review by MauricioNC — APPROVED** [review]

**Review by BlazeDex — APPROVED** [review]

**Pull request merged**